src/Symfony/Component/Config/Definition/Configurator/DefinitionConfigurator.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Config\Definition\Configurator;
+
+use Symfony\Component\Config\Definition\Builder\ArrayNodeDefinition;
+use Symfony\Component\Config\Definition\Builder\NodeDefinition;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\Loader\DefinitionFileLoader;
+
+/**
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ */
+class DefinitionConfigurator
+{
+    public function __construct(
+        private TreeBuilder $treeBuilder,
+        private DefinitionFileLoader $loader,
+        private string $path,
+        private string $file
+    ) {
+    }
+
+    final public function import(string $resource, string $type = null, bool $ignoreErrors = false): void
+    {
+        $this->loader->setCurrentDir(\dirname($this->path));
+        $this->loader->import($resource, $type, $ignoreErrors, $this->file);
+    }
+
+    final public function rootNode(): NodeDefinition|ArrayNodeDefinition
+    {
+        return $this->treeBuilder->getRootNode();
+    }
+
+    final public function setPathSeparator(string $separator): void
```

Comment:
I don't think any of the methods should be final. The reason is that `DefinitionConfigurator` is referenced by `ConfigurationInterface`. Making the methods final closes extensibility in a hard way.

Reply:
I take this approach from `ContainerConfigurator` where every method is final. Should this configurator be treated differently?

Reply:
Yes, because it's part of an interface. Otherwise, the interface is tighly bound to an implementation, which breaks its purpose (abstraction).